idea/src/org/jetbrains/kotlin/idea/inspections/JavaCollectionsStaticMethodCallInspection.kt

```diff
@@ -0,0 +1,104 @@
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.imports.importableFqName
+import org.jetbrains.kotlin.idea.intentions.callExpression
+import org.jetbrains.kotlin.js.descriptorUtils.nameIfStandardType
+import org.jetbrains.kotlin.load.java.descriptors.JavaMethodDescriptor
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+
+class JavaCollectionsStaticMethodInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitDotQualifiedExpression(expression: KtDotQualifiedExpression) {
+                super.visitDotQualifiedExpression(expression)
+
+                val callExpression = expression.callExpression ?: return
+                val context = callExpression.analyze(BodyResolveMode.PARTIAL)
+                val resolvedCall = callExpression.getResolvedCall(context) ?: return
+                val descriptor = resolvedCall.resultingDescriptor as? JavaMethodDescriptor ?: return
+                val fqName = descriptor.importableFqName?.asString() ?: return
+                if (!canReplaceWithStdLib(fqName, callExpression)) return
+
+                holder.registerProblem(expression,
+                                       "Java Collections static method call can be replaced with Kotlin stdlib",
+                                       ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                                       ReplaceWithSortFix(fqName))
+            }
+        }
+    }
+
+    private fun canReplaceWithStdLib(fqName: String, callExpression: KtCallExpression): Boolean {
+        if (!fqName.startsWith("java.util.Collections.")) return false
+        val valueArgs = callExpression.valueArguments
+        val valueArgsSize = valueArgs.size
+        return when (fqName) {
+            "java.util.Collections.reverse" ->
+                valueArgsSize == 1 && isMutableList(valueArgs[0])
+            "java.util.Collections.sort" ->
+                (valueArgsSize == 1 || valueArgsSize == 2) && isMutableList(valueArgs[0])
+            else ->
+                false
+        }
+    }
+
+    private fun isMutableList(arg: KtValueArgument?): Boolean {
+        if (arg == null) return false
+        val expression = arg.getArgumentExpression() ?: return false
+        val context = expression.analyze(BodyResolveMode.PARTIAL)
```

Comment:
You are analyzing the same code twice. You should reuse the `context` from above.